op-chain-ops/cmd/celo-migrate/non-ancients.go

```diff
@@ -21,40 +29,39 @@ func migrateNonAncientsDb(oldDbPath, newDbPath string, numAncients, batchSize ui
 	// Convert output to string
 	outputStr := string(output)
 
-	// TODO(Alec) have rsync run as part of pre-migration (but not the transformation or state)
-	// can use --update and --delete to keep things synced between dbs
-
 	// Check for supported options
 	var cmd *exec.Cmd
 	// Prefer --info=progress2 over --progress
 	if strings.Contains(outputStr, "--info") {
-		cmd = exec.Command("rsync", "-v", "-a", "--info=progress2", "--exclude=ancient", oldDbPath+"/", newDbPath)
+		cmd = exec.Command("rsync", "-v", "-a", "--info=progress2", "--exclude=ancient", "--delete", oldDbPath+"/", newDbPath)
 	} else if strings.Contains(outputStr, "--progress") {
-		cmd = exec.Command("rsync", "-v", "-a", "--progress", "--exclude=ancient", oldDbPath+"/", newDbPath)
+		cmd = exec.Command("rsync", "-v", "-a", "--progress", "--exclude=ancient", "--delete", oldDbPath+"/", newDbPath)
 	} else {
-		cmd = exec.Command("rsync", "-v", "-a", "--exclude=ancient", oldDbPath+"/", newDbPath)
+		cmd = exec.Command("rsync", "-v", "-a", "--exclude=ancient", "--delete", oldDbPath+"/", newDbPath)
 	}
+
+	// rsync copies any file with a different timestamp or size.
+	//
+	// '--exclude=ancient' excludes the ancient directory from the copy
+	//
+	// '--delete' Tells rsync to delete extraneous files from the receiving side (ones that aren’t on the sending side)
+	//
+	// '-a' archive mode; equals -rlptgoD. It is a quick way of saying you want recursion and want to preserve almost everything, including timestamps, ownerships, permissions, etc.
+	// Timestamps are important here because they are used to determine which files are newer and should be copied over.
+	//
+	// '--whole-file' This is the default when both the source and destination are specified as local paths, which they are here (oldDbPath and newDbPath).
```

Comment:
Seems like we don't want to use whole file, otherwise we lose the benefit of rsync (which is not copying things that are already at the destination)

Reply:
--whole-file is used by default when the source and destination are on the same machine (defined as local paths). It increases the amount of data that's copied but because the connection is fast (same machine) I guess it doesn't matter and is optimal vs. finding the diff between each file and copying just the diff which is the default behavior when source and dest are not local paths

Reply:
In any case I did try turing --whole-file off just to test what happens but there doesn't appear to be a way to disable it when source and dest are local paths

Reply:
Actually I think it's fine, I think i misunderstood the docs for rsync.